examples/grpc-server-with-prometheus/server/server.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	"github.com/grpc-ecosystem/go-grpc-prometheus"
+	pb "github.com/grpc-ecosystem/go-grpc-prometheus/examples/grpc-server-with-prometheus/protobuf"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// DemoServiceServer defines a Server.
+type DemoServiceServer struct{}
+
+func newDemoServer() *DemoServiceServer {
+	return &DemoServiceServer{}
+}
+
+// SayHello implements a interface defined by protobuf.
+func (s *DemoServiceServer) SayHello(ctx context.Context, request *pb.HelloRequest) (*pb.HelloResponse, error) {
+	pushCounter.Inc()
+	return &pb.HelloResponse{Message: fmt.Sprintf("Hello %s", request.Name)}, nil
+}
+
+// Create a counter metric
+var (
+	pushCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "repository_pushes",
+		Help: "Number of pushes to external repository.",
+	})
+)
+
+func init() {
+	// Metrics have to be registered to be exposed:
+	prometheus.MustRegister(pushCounter)
+}
+
+func main() {
+	// Listen an actual port.
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", 9093))
```

Comment:
Please defer closing the listener. `grpc.Server.Serve` seems to close listener on return, but lots of fatals can happen until you actually invoke `grpcServer.Serve(lis)`.

Reply:
not addressed.